### ISSUE TITLE
Logj42 ThreadContext - Not visible

### ISSUE THREAD

**Issue by rbeltrand** [rfoltyns/log4j2-elasticsearch#17]
Hi, I set a org.apache.logging.log4j.ThreadContext on my app, and it doesn´t show at kibana, I use the configuration u have on the readme:

```
		<Elasticsearch name="elasticsearchAsyncBatch">
			<IndexName indexName="log4j2" />
			<AsyncBatchDelivery>
				<IndexTemplate name="log4j2"
					path="classpath:indexTemplate.json" />
				<JestHttp serverUris="http://localhost:9200" />
			</AsyncBatchDelivery>
		</Elasticsearch>
```

And in elastic this is the data it register, but I can´t see the info of the threadcontext:
```
{
  "_index": "log4j2",
  "_type": "index",
  "_id": "fCVIQGoBxjbw00kUTTJ1",
  "_version": 1,
  "_score": 0,
  "_source": {
    "timeMillis": 1555856640973,
    "loggerName": "com.myapp.conversion.configuration.LoggerInterceptor",
    "level": "INFO",
    "message": "thread=END",
    "thread": "http-nio-8080-exec-1"
  },
  "highlight": {
    "loggerName": [
      "@kibana-highlighted-field@com.myapp.conversion.configuration.LoggerInterceptor@/kibana-highlighted-field@"
    ]
  }
}
```

This is my log on console (<PatternLayout pattern="%d{yyyy-MM-dd HH:mm:ss} %-5p (%c{1}.java:%L)  => %m %X%n" />):
`2019-04-21 09:24:00 INFO  (LoggerInterceptor.java:47)  => thread=END {execution-duration=733, fusionId=123, path=/DataConvert/v2/operacion/123, threadId=89cbe3be-d7f9-4977-8e15-0af1632c6dc2, userIp=0:0:0:0:0:0:0:1}`

**Comment by rfoltyns** [comment]
@rbeltrand `ContextData` is not serialized by default - it's ignored in `LogEventJacksonJsonMixIn`. You can:
- override this MixIn as described [here](https://github.com/rfoltyns/log4j2-elasticsearch/issues/16#issuecomment-475624291)
- or use regular `JsonLayout` if you're not using object pooling

Support for `ContextData` can be added in future releases.

**Comment by rfoltyns** [comment]
@rbeltrand Please, visit #18 for a more comprehensive solution. Hope it helps.

**Comment by rfoltyns** [comment]
`VirtualProperty` released in 1.4.
